tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,599 @@
+
+  void setAllowedFormats(nvinfer1::TensorFormats formats) {}
+
+  nvinfer1::TensorFormats getAllowedFormats() const { return 1; }
+
+  bool isShapeTensor() const { return false; }
+  bool isExecutionTensor() const { return true; }
+
+ private:
+  nvinfer1::DataType trt_dtype_;
+  nvinfer1::Dims trt_dims_;
+  std::string name_;
+  nvinfer1::TensorLocation location_;
+  float dynamic_range_;
+};
+
+// Fake ITensor implementation for testing purposes.
+class FakeITensor {
+ public:
+  FakeITensor() : dynamic_range_(0.0f) {}
+
+  FakeITensor(const nvinfer1::Dims& dims) : dims_(dims), dynamic_range_(0.0f) {}
+
+  FakeITensor(const std::vector<int>& dims)
+      : dims_(GetTestDims(dims)), dynamic_range_(0.0f) {}
+
+  void setName(const char* name) { name_ = name; }
+
+  const char* getName() const { return name_.c_str(); }
+
+  void setDimensions(nvinfer1::Dims dimensions) { dims_ = dimensions; }
+
+  nvinfer1::Dims getDimensions() const { return dims_; }
+
+  void setType(nvinfer1::DataType type) { type_ = type; }
+
+  nvinfer1::DataType getType() const { return type_; }
+
+  bool isNetworkInput() const { return false; }
+
+  bool isNetworkOutput() const { return false; }
+
+  void setBroadcastAcrossBatch(bool broadcastAcrossBatch) {}
+
+  bool getBroadcastAcrossBatch() const { return false; }
+
+  nvinfer1::TensorLocation getLocation() const { return location_; }
+
+  void setLocation(nvinfer1::TensorLocation location) {
+    location_ = location;
+  }
+
+  bool setDynamicRange(float min, float max) {
+    dynamic_range_ = std::max(std::abs(min), std::abs(max));
+    return true;
+  }
+
+  bool dynamicRangeIsSet() const { return true; }
+
+  void resetDynamicRange() {}
+
+  float getDynamicRangeMin() const { return 0.f; }
+
+  float getDynamicRangeMax() const { return 0.f; }
+
+  void setAllowedFormats(nvinfer1::TensorFormats formats) {}
+
+  nvinfer1::TensorFormats getAllowedFormats() const { return 1; }
+
+  bool isShapeTensor() const { return false; }
+  bool isExecutionTensor() const { return true; }
+
+ private:
+  std::string name_;
+  nvinfer1::Dims dims_;
+  nvinfer1::DataType type_;
+  nvinfer1::TensorLocation location_;
+  float dynamic_range_;
+};
+
+enum class TensorType : int
+{
+  kNONE,
+  kTRT,
+  kSIMPLE,
+  kFAKE
+};
+
+class ITensorProxy
+{
+public:
+  //! ITensor not owned
+  ITensorProxy(nvinfer1::ITensor* trt_tensor)
+    : trt_tensor_(trt_tensor),
+      ttype_(TensorType::kTRT) {}
+
+  //! SimpleITensor owned
+  ITensorProxy(SimpleITensor* simple_itensor)
+    : simple_tensor_(simple_itensor),
+      ttype_(TensorType::kSIMPLE) {}
+
+  //! FakeITensor owned
+  ITensorProxy(FakeITensor* fake_tensor)
+    : fake_tensor_(fake_tensor),
+      ttype_(TensorType::kFAKE) {}
+    
+  //! SimpleITensor owned
+  explicit ITensorProxy(nvinfer1::DataType trt_dtype, const nvinfer1::Dims& trt_dims)
+    : simple_tensor_(std::unique_ptr<SimpleITensor>(new SimpleITensor(trt_dtype, trt_dims))),
+      ttype_(TensorType::kSIMPLE) {}
```

Comment:
This code can't be compiled. Please note that the order of the class member initialization needs to be the same as the order of the class member declaration. Similar in a few other constructors here.

Reply:
Good catch, I've moved the tensor type to be the last declared member of the class which looks to take care of the issue